sanic/response.py

```diff
@@ -73,28 +73,121 @@
 }
 
 
-class HTTPResponse:
+class BaseHTTPResponse:
+    def _encode_body(self, data):
+        try:
+            # Try to encode it regularly
+            return data.encode()
+        except AttributeError:
```

Comment:
@seemethere I get what @r0fls is saying, the body being passed into the encoder might already be in bytes form (like if it is jsonified) so a type check is faster than attempting to re-encode it. I'll add it in.